tools/snaplink-sherlock.sh

```diff
@@ -49,11 +50,19 @@ exec 4>>$xtrace_log
 BASH_XTRACEFD=4
 set -o xtrace
 
+# Parse the full moray shard out of the postgres zone alias.
+# E.g.:
+#       1.postgres.coalregion.joyent.us-f8bd09a5
+#       {shardnum}.postgres.{region}.{dns_domain}-{instance_prefix}
+# from which (assuming always that shard="{shardnum}.moray"):
+#       shard=1.moray.{region}.{dns_domain}
 targetAlias=$(vmadm get ${TARGET} | json -H alias)
-[[ ${targetAlias} =~ ^[0-9]+\.postgres\. ]] || fatal "VM '${targetAlias}' does not look like a Manta manatee zone."
-shard=$(cut -d'.' -f1 <<<${targetAlias})
+[[ ${targetAlias} =~ ^[0-9]+\.postgres\.[a-z\.-]+-[0-9a-f]+ ]] \
```

Comment:
need to relax the regex for region name check to `[0-9a-f\.-]` to avoid:
```
VM '1.postgres.qaserver3.scloud.host-09808b3d' does not look like a Manta postgres zone.
```
(though we don't have such naming convention for region in our clouds)

Reply:
Hrm, okay. Note that this after-the-fact change to prompt-config.sh https://github.com/joyent/sdc-headnode/commit/babd36cf798e808633e8e86178082459b8f78a7d  makes it so "." is disallowed in the region_name

Reply:
Allowing `.` in the region_name seems like a bug in the first place. Isn't it?

Reply:
^^ I misquoted the allowed chars there. I'll have an update in a little bit.

Reply:
My bad. I missed the `0-9` range in the region_name and dns_domain segments:

```diff
-[[ ${targetAlias} =~ ^[0-9]+\.postgres\.[a-z\.-]+-[0-9a-f]+ ]] \
+[[ ${targetAlias} =~ ^[0-9]+\.postgres\.[a-z0-9\.-]+-[0-9a-f]+ ]] \
```

which works to validate `1.postgres.qaserver3.scloud.host-09808b3d`.   It worked in my testing because my regions were "coalregion" and "nightly" (no numbers).

Updated.